myjsonrpc.pm

```diff
@@ -48,36 +63,47 @@ our $sockets;
 sub read_json {
     my ($socket, $cmd_token) = @_;
 
-    my $cjx = Cpanel::JSON::XS->new;
 
     my $fd = fileno($socket);
+    if ($DEBUG_JSON) {
+        bmwqemu::diag("($$) read_json($fd)");
+    }
+    my $buffer = '';
     if (exists $sockets->{$fd}) {
         # start with the trailing text from previous call
-        $cjx->incr_parse($sockets->{$fd});
-        delete $sockets->{$fd};
+        $buffer = delete $sockets->{$fd};
     }
 
     my $s = IO::Select->new();
     $s->add($socket);
 
-    my $hash;
+    my $multi = wantarray ? 1 : 0;
```

Comment:
I'd prefer passing the `$multi` flag as argument (or even make it multiple functions).